### PR TITLE
gitignore: add gitignore

### PR DESCRIPTION
**Pull request by yunmingyang**
Add .idea and .vscode to toplevel
Add  test/static-code to subdirs

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -22,6 +22,8 @@
 /pkg/
 /tmp/
 /tools/
+/.idea
+/.vscode
```
Comment (martinpitt) [inline .gitignore:26]: Hmm, this rather belongs into your global config, doesn't it? Adding every editor's temporary files to every project is too much clutter IMHO.

Comment (yunmingyang) [reply .gitignore:26]: OK, then I will remove them

File: .gitignore [hunk 2]
```diff
@@ -30,3 +32,4 @@
 /po/LINGUAS
 /test/common/
 /test/images/
+/test/static-code
```
Comment (skobyda) [inline .gitignore:35]: Why would this be part of gitignore? It's cockpit's custom code there which we might want to update in the future for the whole repo.

File: .gitignore (hunk 1 above)
Comment (skobyda) [reply .gitignore:26]: I agree that this should be part of each developer's global config. There then is much more IDEA and OS folders to be gitignored specific for each developer, such as .DS_Store, and I don't feel like getting into this rabbit hole

File: .gitignore (hunk 2 above)
Comment (skobyda) [reply .gitignore:35]: Ah right, this should be mantained in cockpit's repo, and not c-machines. Then I agree that this can be gitignored.

**Review by martinpitt — APPROVED** [review]
Thanks!

**Pull request merged**